File: 02.04.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val) { $this->val = $val; }
 * }
 */
class Solution {

    /**
     * @param ListNode $head
     * @param Integer $x
     * @return ListNode
     */
    function partition($head, $x) {
        if (!$head || !$head->next) {
            return $head;
        }

        $cur = $head;
        $p = $head;
        while ($cur) {
            if ($cur->val < $x) {
                $temp = $p->val;
                $p->val = $cur->val;
                $cur->val = $temp;
                $p = $p->next;
            }
            $cur = $cur->next;
        }

        return $head;
    }
}

File: go/linkedList/02.04.go
package main

//type ListNode struct {
//	Val  int
//	Next *ListNode
//}
//
//func partition(head *ListNode, x int) *ListNode {
//	if head == nil || head.Next == nil {
//		return head
//	}
//
//	cur, pre := head, head
//
//	for cur != nil {
//		if cur.Val < x {
//			cur.Val, pre.Val, pre = pre.Val, cur.Val, pre.Next
//		}
//		cur = cur.Next
//	}
//
//	return head
//}
